### DISCUSSION TITLE
Public 2.7.4 api documentation

### DISCUSSION THREAD

**Discussion by cfrickey** [netbox-community/netbox#10484] [Q&A] [answered]
We have an instance of netbox version 2.7.4 installed on our internal network.  We are now working with an external vendor who needs the api documentation, but they do not have access to our network to view the documentation from our installation.  We have done some customization and are not yet in a position to update to the latest version, so the documentation on the demo or readthedocs will be outdated to our implementation.  Is there somewhere that archive documentation is available on the public web?  I have done some searching but found nothing yet.

Thanks in advance!

**Comment by candlerb** [comment, marked as answer]
1. https://github.com/netbox-community/netbox/tree/v2.7.4/docs
2. `curl http://localhost:8001/api/docs/?format=openapi >netbox.openapi.json` and give this file to your external developers; they can open it an OpenAPI tool of their choice.

However, be very wary going down this path.  Netbox API changes from version to version; if your external developers develop to v2.7, then they'll have to make substantial changes to their work later when you upgrade.  I think it's better to get your upgrades out of the way first.

As for your local modifications to Netbox: depending on what you've done, this may be hard for you to unpick.  However, if you can take your work and turn it into a Netbox "[plugin](https://docs.netbox.dev/en/stable/plugins/development/)" (first introduced in v2.8) then it should be easier to maintain separately.

**Reply by cfrickey** [reply]
Thank you so much!

I totally agree with the development path.  Unfortunately, the new dev is outside of my department, so I have somewhat limited influence there, though I am definitely going to share my concerns.

I don't know that we really have that many local modifications to Netbox code, but we will have some fairly substantial process impacts (generally for the better) in the upgrade.  I want to make sure we plan our upgrade to mitigate end-user pain.
